packages/gatsby/src/bootstrap/load-plugins/load.js

```diff
@@ -89,17 +89,20 @@ function resolvePlugin(pluginName, rootDir) {
         ? createRequireFromPath(`${rootDir}/:internal:`)
         : require
     const resolvedPath = slash(path.dirname(requireSource.resolve(pluginName)))
+    const pluginNameIsPath = path.isAbsolute(pluginName)
+    const packageJSONPath = pluginNameIsPath
+      ? `${resolvedPath}/package.json`
+      : requireSource.resolve(`${pluginName}/package.json`)
```

Comment:
I suspect the right fix would be to make sure `resolvedPath` points to the base package directory (containing `package.json`) and not to the directory containing the main `index.js` file.

Then we don't need `packageJSONPath` (and don't need to modify `internal-data-bridge`).

Reply:
Few notes from our discussion about this:

> @Vladar:
> With current PR it tries loading gatsby-node from <PLUGIN_ROOT>/lib/gatsby-node. With my suggestion - searches  in <PLUGIN_ROOT>/gatsby-node so yeah - it's about what we consider the right thing

This is the question we need to answer. In my opinion it's not very intuitive that for us to use `<PLUGIN_ROOT>/lib/gatsby-node` - we are setting `main`, but `gatsby-node` is not really `main`.

I would lean toward using @vladar's suggestion for now. This will also match docs we have about plugin structure: https://www.gatsbyjs.org/docs/files-gatsby-looks-for-in-a-plugin/ (we look for gatsby-X files in the root of package).

If there will be need to support `gatsby-node` in different location we should provide proper way of doing that (maybe custom package.json fields similar to `main` for `gatsby-X` files)

Thoughts?

Reply:
Makes sense to me